<?xml version="1.0" ?>
<launch>    
  <node pkg="tf2_ros" type="static_transform_publisher" name="camera1_broadcaster" args="-0.030588 -0.072660 0.3 -0.679470685101 0.67950216353 -0.195694723206 0.195703789325 /world /camera_link_optical" />     

  <arg name="rviz"  default="true"/>
  <!-- marker detector -->
  <!-- aruco_ros node is used to detect markers -->
  <!-- ros wiki page: http://wiki.ros.org/aruco_ros -->
  <!-- external page: http://www.uco.es/investiga/grupos/ava/node/26 -->  
  
  <!-- camera info, image and camera_frame are taken from default values of aruco_single.launch-->
  <include file = "$(find gazebo_sensors_tutorial)/launch/aruco_single.launch.xml">
    <arg name="node_name"    value="aruco_single_582"/> 
    <arg name="markerId"     value="582"/> 
    <arg name="markerSize"   value="0.045"/>  
    <arg name="marker_frame" value="aruco_582"/>
    <arg name="reference_frame"    value="world"/>  
  </include>

  <!-- camera info, image and camera_frame are taken from default values of aruco_single.launch-->
  <include file = "$(find gazebo_sensors_tutorial)/launch/aruco_single.launch.xml" >    
    <arg name="node_name"    value="aruco_single_201"/> 
    <arg name="markerId"     value="201"/>  
    <arg name="marker_frame" value="aruco_201"/>
    <arg name="markerSize"   value="0.03"/>  
    <arg name="reference_frame"    value="world"/>  
  </include>
  
  <node pkg="gazebo_sensors_tutorial" type="aruco_frames_detection" name="aruco_frames_detection"/>
  
<!--
  <arg name="markerSize"      default="0.03"/> 
  <arg name="referenceFrame"       default=""/>  
  <arg name="cameraInfo"     default="/camera1/camera_info"/>
  <arg name="image"           default="/camera1/image_rect_color"/>
  <arg name="cameraFrame"    default="/camera_link_optical"/>
  
  <node pkg="aruco_ros" type="marker_publisher" name="aruco_marker_publisher">
    <remap from="/camera_info" to="$(arg cameraInfo)" />
    <remap from="/image" to="$(arg image)" />
    <param name="image_is_rectified" value="True"/>
    <param name="marker_size"        value="$(arg markerSize)"/>
    <param name="reference_frame"    value="$(arg referenceFrame)"/>   
    <param name="camera_frame"       value="$(arg cameraFrame)"/>
  </node>
  -->

  <!-- image_proc node sits between the camera driver and vision processing nodes. -->
  <!-- image_proc removes camera distortion from the raw image stream -->
  <!-- requires to set the ROS_NAMESPACE to the camera publishing the images, camera1 in our case -->
  <!-- using:
         <env name="ROS_NAMESPACE" value="camera1" /> 
         <node name="image_proc" pkg="image_proc" type="image_proc" />
       or the ns inside the call to the node image_proc
  -->
  <!-- See http://wiki.ros.org/image_proc for more info -->
  <node ns="camera" name="image_proc" pkg="image_proc" type="image_proc" />
     
  <!-- Show in Rviz   -->   
  <node name="rviz" pkg="rviz" type="rviz" args="-d $(find gazebo_sensors_tutorial)/config/camera_aruco_cube-2.rviz"/> 
</launch>

